cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSink.java

```diff
@@ -71,6 +76,29 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   protected SchemaReader getSchemaReader() {
     return new CommonSchemaReader();
   }
+
+  @Override
+  protected LineageRecorder getLineageRecorder(BatchSinkContext context) {
+    String host;
+    String location = "";
+    if (CloudSQLUtil.PRIVATE_INSTANCE.equalsIgnoreCase(cloudsqlMysqlSinkConfig.getConnection().getInstanceType())) {
+      // connection is the private IP address
+      host = cloudsqlMysqlSinkConfig.getConnection().getConnectionName();
+    } else {
+      // connection is of the form <projectId>:<region>:<instanceName>
+      String[] connectionParams = cloudsqlMysqlSinkConfig.getConnection().getConnectionName().split(":");
+      host = connectionParams[2];
+      location = connectionParams[1];
+    }
+    String fqn = DBUtils.constructFQN("mysql", host, 3306,
+                                      cloudsqlMysqlSinkConfig.getConnection().getDatabase(),
+                                      cloudsqlMysqlSinkConfig.getReferenceName());
+    Asset.Builder assetBuilder = Asset.builder(cloudsqlMysqlSinkConfig.getReferenceName()).setFqn(fqn);
+    if (!StringUtils.isEmpty(location)) {
```

Comment:
So if I do not initialize location the IDE complains. It doesn't happen with host because it is initialized in if and else clauses. This means in one place or the other we will have to initialize it